app/services/task_service/gpt_generate_tests.rb

```diff
@@ -2,11 +2,14 @@
 
 module TaskService
   class GptGenerateTests < ServiceBase
-    def initialize(task:)
+    def initialize(task:, user:)
       super()
       raise Gpt::MissingLanguageError if task.programming_language&.language.blank?
 
       @task = task
+      @user = user
```

Comment:
Why did you decide to pass the `User` instead of the API-Key itself? I would prefer the latter. Either way you are assigning `@user` here and don't use it anywhere outside the initialize, so you could just inline it in the line below.

Reply:
becasue the user might not have an API key available. so it made sense to enter the user instead of the key. is that wrong?

Reply:
I would still just pass the API-Key and check for `present?`. The service doesn't care about any other information from the `User` but the Key.